### PR TITLE
feat(shorebird_cli): add `shorebird upgrade`

### PR DESCRIPTION
**Pull request by felangel**
## Description

- feat(shorebird_cli): add shorebird upgrade (part of #107)

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/upgrade_command.dart [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      logger.err('Fetching current version failed: ${error.message}');
+      return ExitCode.software.code;
+    }
+
+    late final String latestVersion;
+    try {
+      latestVersion = await fetchLatestVersion(
+        workingDirectory: workingDirectory,
+      );
+    } on ProcessException catch (error) {
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/upgrade_command.dart:44]: You're going to delete so much code when you make this on ProcessException into some sort of function wrapper thing. 🤣

File: packages/shorebird_cli/lib/src/commands/upgrade_command.dart [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  /// Returns the local HEAD flutter version.
+  ///
+  /// Exits tool if HEAD isn't pointing to a branch, or there is no upstream.
+  Future<String> fetchCurrentVersion({
+    required String workingDirectory,
+  }) async {
+    // Get the commit revision of HEAD
+    final result = await runProcess(
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/upgrade_command.dart:105]: Same idea, this repeated runProcess behavior probably ends up a function. 🤷

**Review by eseidel — APPROVED** [review]
Looks great!  Land as-is lgtm.

File: packages/shorebird_cli/lib/src/commands/upgrade_command.dart (hunk 1 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/commands/upgrade_command.dart:44]: Yeah we really need to do that 😅

**Pull request merged**